src/react/hooks/use-room.ts

```diff
@@ -0,0 +1,128 @@
+import { ConnectionStatusChange, Room, RoomLifecycle, RoomStatusChange } from '@ably/chat';
+import * as Ably from 'ably';
+import { useCallback, useContext, useEffect, useState } from 'react';
+
+import { ChatStatusResponse } from '../chat-status-response.js';
+import { RoomContext } from '../contexts/room-context.js';
+import { useChatConnection } from './use-chat-connection.js';
+
+/**
+ * Parameters for the {@link useRoom} hook.
+ */
+export interface UseRoomParams {
+  /**
+   * Callback for when the room status changes. The listener is removed when
+   * the component unmounts.
+   * @param change The change object.
+   */
+  onStatusChange?: (change: RoomStatusChange) => void;
+
+  /**
+   * Callback for when the connection status changes. The listener is removed
+   * when the component unmounts.
+   * @param change The change object.
+   */
+  onConnectionStatusChange?: (change: ConnectionStatusChange) => void;
+}
+
+/**
+ * The return type for the {@link useRoom} hook.
+ */
+export interface UseRoomResponse extends ChatStatusResponse {
+  /** The room object. */
+  room: Room;
+
+  /**
+   * Shortcut to {@link Room.attach}. Not needed if the {@link RoomProvider}
+   * has `attach=true` set, which is the default.
+   */
+  attach: () => Promise<void>;
+
+  /**
+   * Shortcut to {@link Room.detach}. Not needed if the {@link RoomProvider}
+   * has `attach=true` or `release=true` set, which are the default values.
+   */
+  detach: () => Promise<void>;
+}
+
+/**
+ * Helper function to create a status object from a change event or the room
+ * status. It makes sure error isn't set at all if it's undefined in the source.
+ */
+function makeStatusObject(source: { current: RoomLifecycle; error?: Ably.ErrorInfo }) {
+  if (source.error) {
+    return {
+      status: source.current,
```

Comment:
I'm not sure I understand the purpose of this `if` statement, the two appear identical?

Reply:
I'm not too sure, it looks equal to me too, I think I'll just remove the conditional block and see if anything breaks.